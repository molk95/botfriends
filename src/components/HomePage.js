import React, { Component } from "react";

import CardList from "../components/CardList";
import FilterBots from "../components/FilterBots";
import Loading from "../components/Loading";
import "./HomePage.css";

import CounterButton from "../components/CounterButton";

class HomePage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }
  filterRobots = robotFriends => {
    return robotFriends.filter((robot) => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    })
  }
  render() {
    const { searchField, onSearchChange, robotFriends, isPending } = this.props;
    return !isPending ? (
      <Loading />
    ) : (
      <div className="tc">
        <h1 className="f1">botFriends</h1>
        <FilterBots searchChange={onSearchChange} />
        <CounterButton />
        <CardList
          robotFriends={this.filterRobots(robotFriends)}
        />
      </div>
    );
  }
}
export default HomePage;
