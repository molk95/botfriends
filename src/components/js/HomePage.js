import React, { Component } from "react";

import CardList from "../js/CardList";
import FilterBots from "./FilterBots";
import Loading from "../js/Loading";
import "../css/HomePage.css";



class HomePage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }
  filterRobots = (robotFriends) => {
    return robotFriends.filter((robot) => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };
  render() {
    const { onSearchChange, robotFriends, isPending } = this.props;
    return !isPending ? (
      <Loading />
    ) : (
      <div className="tc">
        <h1 className="f1">botFriends</h1>
        <FilterBots searchChange={onSearchChange} />
        <CardList robotFriends={this.filterRobots(robotFriends)} />
      </div>
    );
  }
}
export default HomePage;
