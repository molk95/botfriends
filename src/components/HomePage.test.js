import { shallow } from "enzyme";
import React from "react";
import HomePage from "./HomePage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robotFriends: [],
    searchFiled: "",
    isPending: false,
  };
  wrapper = shallow(<HomePage {...mockProps} />);
});

it("renders HomePage without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

it("filters robots correctly", () => {
  const mockPropsfilter = {
    onRequestRobots: jest.fn(),
    robotFriends: [
      {
        id: 4,
        name: "jane",
        email: "jane@xyz.io",
      },
    ],
    searchFiled: "jane",
    isPending: false,
  };
  const wrapperFilter = shallow(<HomePage {...mockPropsfilter} />);
  expect(wrapperFilter.instance().filterRobots()).toEqual([
    {
      id: 4,
      name: "jane",
      email: "jane@xyz.io",
    },
  ]);
});
