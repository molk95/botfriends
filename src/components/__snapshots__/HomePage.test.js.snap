// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage without crashing 1`] = `ShallowWrapper {}`;
